File: app/src/main/java/vn/loitp/app/activity/customviews/progress_loadingview/circularprogressbar/CircularProgressBarActivity.java
package vn.loitp.app.activity.customviews.progress_loadingview.circularprogressbar;

import android.app.Activity;
import android.os.Bundle;

import vn.loitp.app.activity.customviews.progress_loadingview.circularprogressbar.lib.CircularProgressBar;
import vn.loitp.app.base.BaseActivity;
import vn.loitp.livestar.R;

public class CircularProgressBarActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        CircularProgressBar progressBar = findViewById(R.id.progress_bar);
        progressBar.setProgress(30f);
        progressBar.configure().animateProgress(true).maximum(40).progress(30).apply();
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected Activity setActivity() {
        return this;
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_circular_progress_bar;
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/progress/circularProgressBar/CircularProgressBarActivity.kt
package vn.loitp.app.activity.customviews.progress.circularProgressBar

import android.os.Bundle
import androidx.core.view.isVisible
import com.loitp.annotation.IsFullScreen
import com.loitp.annotation.LogTag
import com.loitp.core.base.BaseFontActivity
import com.loitpcore.core.utilities.LUIUtil
import kotlinx.android.synthetic.main.activity_progress_circular_progress_bar.*
import vn.loitp.app.R

@LogTag("CircularProgressBarActivity")
@IsFullScreen(false)
class CircularProgressBarActivity : BaseFontActivity() {

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_progress_circular_progress_bar
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setupViews()
    }

    private fun setupViews() {
        lActionBar.apply {
            LUIUtil.setSafeOnClickListenerElastic(
                view = this.ivIconLeft,
                runnable = {
                    onBaseBackPressed()
                }
            )
            this.ivIconRight?.isVisible = false
            this.tvTitle?.text = CircularProgressBarActivity::class.java.simpleName
        }

        progressBar.progress = 30f
        progressBar.configure()
            .animateProgress(true)
            .maximum(40f)
            .progress(30f)
            .apply()
    }
}
